app/assets/javascripts/discourse/tests/acceptance/composer-test.js

```diff
@@ -1394,3 +1394,48 @@ acceptance("Composer - current time", function (needs) {
     );
   });
 });
+
+acceptance("composer buttons API", function (needs) {
+  needs.user();
+  needs.settings({
+    allow_uncategorized_topics: true,
+  });
+
+  test("buttons can be added conditionally", async function (assert) {
+    withPluginApi("0", (api) => {
+      api.addComposerToolbarPopupMenuOption({
+        action: (toolbarEvent) => {
+          toolbarEvent.applySurround("**", "**");
+        },
+        icon: "far-bold",
+        label: "some_label",
+        condition: (composer) => {
+          return composer.model.creatingTopic;
+        },
+      });
+    });
+
+    await visit("/t/internationalization-localization/280");
+
+    await click("#topic-footer-buttons .btn.create");
```

Comment:
Somewhat unrelated, the `.btn.create` selector doesn't communicate "this is a reply button" to me. We do use `.reply.create` on reply buttons in the post buttons.